niap-cc/KeyTestingPatches/Android16/frameworks/base/services/core/java/com/android/server/locksettings/SyntheticPasswordCrypto.java.patch

```diff
@@ -0,0 +1,83 @@
+--- SyntheticPasswordCrypto.java.orig	2025-11-17 03:56:58.255075545 +0000
++++ SyntheticPasswordCrypto.java	2025-11-17 19:44:21.468945049 +0000
+@@ -49,8 +49,13 @@
+ import javax.crypto.spec.GCMParameterSpec;
+ import javax.crypto.spec.SecretKeySpec;
+ 
++import android.util.Log;
++import java.io.UnsupportedEncodingException;
++
+ class SyntheticPasswordCrypto {
+     private static final String TAG = "SyntheticPasswordCrypto";
++    private static final String TAGKMD = "SyntheticPasswordCrypto(KMD)";
++
+     private static final int AES_GCM_KEY_SIZE = 32; // AES-256-GCM
+     private static final int AES_GCM_IV_SIZE = 12;
+     private static final int AES_GCM_TAG_SIZE = 16;
+@@ -254,21 +259,65 @@
+     protected static byte[] personalizedHash(byte[] personalization, byte[]... message) {
+         try {
+             final int PADDING_LENGTH = 128;
++            
++            StringBuilder logMessage = new StringBuilder();
++            logMessage.append("personalizedHash-> ");
++
+             MessageDigest digest = MessageDigest.getInstance("SHA-512");
+             if (personalization.length > PADDING_LENGTH) {
+                 throw new IllegalArgumentException("Personalization too long");
+             }
+             // Personalize the hash
+             // Pad it to the block size of the hash function
++            logMessage.append(" personalization: ");
++            logMessage.append(new String(personalization, "UTF-8"));
+             personalization = Arrays.copyOf(personalization, PADDING_LENGTH);
+             digest.update(personalization);
++            logMessage.append(" message: [");
+             for (byte[] data : message) {
+                 digest.update(data);
+             }
+-            return digest.digest();
++            logMessage.append(" ]");
++            
++            byte[] res = digest.digest();
++            logMessage.append(" digest: ");
++            logMessage.append(bytesToHex(res));
++            Log.e(TAGKMD, logMessage.toString());
++
++            return res;
+         } catch (NoSuchAlgorithmException e) {
+             throw new IllegalStateException("NoSuchAlgorithmException for SHA-512", e);
++        } catch (UnsupportedEncodingException e) {
++            throw new IllegalStateException("Unable to represent bytes as UTF-8", e);
++        }
++    }
++
++   /**
++    * Uppercase hex string for byte array
++    */
++   public static String bytesToHex(byte[] bytes) {
```

Comment:
Is this method used anywhere else aside from `personalizedHash` above? If that's the only use, can the call to `new String(bytesToHexBytes(bytes), "UTF-8")` be inlined in place of the call to this method above? It looks like the above method already has a catch block for an `UnsupportedEncodingException` that is caught here, but when it's caught here, it is replaced by a `RuntimeException`, so the `IllegalStateException` that is thrown above is never returned to the user. Is this working as intended, or should the `IllegalStateException` ultimately be what is thrown in the method above?

Reply:
Actually I copied and pasted the existing source about this section.
 
But we should't use this method if the byte array represent the String object.
The first argument in the personalizedHash() is it. 
We use the Byte array representation of the string as a part of the digest of the encryption, in this case.

So the catch block for UnsupportedEncodingException  in the method is not redundant.